students/tim_lurvey/lesson05/mailroom_pt3.py

```diff
@@ -0,0 +1,151 @@
+    # one-liners.
+    # "\n".join(["The following names are available:",] + [name for name in my_data.keys()])
+    print_list = "The following names are available:\n"
+    for i, item in enumerate(my_data.items()):
+        print_list += "{i:>3} : {name}\n".format(i=i, name=item[0])
+    print(print_list[:-1])
+    return True
+
+
+def send_thank_you():
+    """This method will send a thank you notice to a user in the database
+    who has made a new donation."""
+    request = input('Enter a full name or "list" to view names\n>>> ')
+    #
+    if 'list' == request.strip().lower():
+        print_name_list()
+        return True
+    else:
+        # add the name is it is not in the database
+        if request not in my_data:
+            add_new_name(name=request)
+        # get the donation amount
+        donation = input("Enter amount:").replace("$", "").strip()
+        # if the number isn't castable to a float, warn user and exit function.
+        try:
+            float(donation)
+        except ValueError:
+            print("The value you entered is not a valid donation.  Error: '{}'\n>>> ".format(donation))
+            return True
+        # add the donation to their existing amount
+        add_donation(name=request, ammount=float(donation))
+        # return the email string
+        print(compose_email(name=request, donation=float(donation), data=my_data.get(request)))
+        return True
+
+
+def report():
+    """ Print report in the following format:
+
+    Donor Name                | Total Given | Num Gifts | Average Gift
+    ------------------------------------------------------------------
+    William Gates, III         $  653784.49           2  $   326892.24"""
+    s = "\n"
+    s += "Donor Name                | Total Given | Num Gifts | Average Gift\n"
+    s += ("-" * 66) + "\n"
+    for i, data in enumerate(my_data.items()):
+        total = "$ {:.2f}".format(data[1].total)
+        average = "$ {:.2f}".format(data[1].total / data[1].number)
+        s += "{name:<26}|{tot:>13}|{num:>11d}|{avg:>13}\n".format(name=data[0],
+                                                                  tot=total,
+                                                                  num=data[1].number,
+                                                                  avg=average, )
+    s += ("-" * 66) + "\n"
+    print(s)
+    return True
+
+
+def fquit():
+    """Exit the application by unseating the switch"""
+    print("\nGoodbye! Exiting program...")
+    return False
+
+
+def error(inpt: str = ""):
+    """Report an error"""
+    print("\nError on input.  Invalid selection \"{}\"".format(inpt))
+    return True
+
+
+def main(args):
+    """This is the controlling logic for the program.  The main loop will
+    repeat forever until the user specifies to quit."""
+    # this is the switch variable
+    CONTINUE = True
+    # while switch is true, loop
+    while CONTINUE:
```

Comment:
this should just be a while True loop and use all caps implies a constant (which is isn't)